module-check/src/main/clojure/clojure/core/typed/base_env.clj

```diff
@@ -1443,7 +1443,17 @@ clojure.core/rand-nth (All [x] [(U (Indexed x) (SequentialSeqable x)) -> x])
   this-ns
 clojure.set/union (All [x] [(Set x) * -> (Set x)])
 clojure.set/intersection (All [x] [(Set x) (Set x) * -> (Set x)])
+
 clojure.set/difference (All [x] [(Set x) (Set Any) * -> (Set x)])
+clojure.set/subset? [(Set Any) (Set Any) -> Boolean]
+clojure.set/superset? [(Set Any) (Set Any) -> Boolean]
+clojure.set/join [(Set (Map Any Any)) (Set (Map Any Any)) -> (Set (Map Any Any))]
```

Comment:
`join` has 2 versions, one takes 2 arguments and another takes three. You should add annotations for both. You can start by adding correct testcase for three-arg version and check that it fails and then update annotation and then the new test case should pass. 
Example of 3-arg version might be:
```clojure
(join #{{:name "betsy" :kind "cow"}} #{{:species "cow" :personality :stoic}} {:kind :species})
```